### ISSUE TITLE
ocr_detection: no output and error with demo_vis.py

### ISSUE THREAD

**Issue by ianni67** [NVlabs/prismer#20]
When running demo_vis.py, I got the following error:
`FileNotFoundError: [Errno 2] No such file or directory: 'helpers/labels/ocr_detection/helpers/images/COCO_test2015_000000000014_.png'`
I found that ocr_detection never outputs anything, but does not raise any error or warning.
Can you advise, please?

**Comment by lorenmt** [comment]
Hello. That doesn't look right. What about other experts?

**Comment by ianni67** [comment]
Hi, the other experts work properly.

**Comment by lorenmt** [comment]
Could you send the error messages or anything so I can look into the details? 

Anyways, if you just want to play with the models, you can also try the official demo on huggingface.

**Comment by ianni67** [comment]
Yes of course.
This is what I get when running demo_vis.py:

```
~/SW/prismer$ python demo_vis.py
Traceback (most recent call last):
  File "demo_vis.py", line 128, in <module>
    ocr_detection_prettify(im_list[i])
  File "demo_vis.py", line 118, in ocr_detection_prettify
    plt.savefig(save_path, bbox_inches='tight', transparent=True, pad_inches=0)
  File "/home/ianni/.virtualenvs/PRISMER/lib/python3.8/site-packages/matplotlib/pyplot.py", line 1023, in savefig
    res = fig.savefig(*args, **kwargs)
  File "/home/ianni/.virtualenvs/PRISMER/lib/python3.8/site-packages/matplotlib/figure.py", line 3343, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/home/ianni/.virtualenvs/PRISMER/lib/python3.8/site-packages/matplotlib/backend_bases.py", line 2366, in print_figure
    result = print_method(
  File "/home/ianni/.virtualenvs/PRISMER/lib/python3.8/site-packages/matplotlib/backend_bases.py", line 2232, in <lambda>
    print_method = functools.wraps(meth)(lambda *args, **kwargs: meth(
  File "/home/ianni/.virtualenvs/PRISMER/lib/python3.8/site-packages/matplotlib/backends/backend_agg.py", line 509, in print_png
    self._print_pil(filename_or_obj, "png", pil_kwargs, metadata)
  File "/home/ianni/.virtualenvs/PRISMER/lib/python3.8/site-packages/matplotlib/backends/backend_agg.py", line 458, in _print_pil
    mpl.image.imsave(
  File "/home/ianni/.virtualenvs/PRISMER/lib/python3.8/site-packages/matplotlib/image.py", line 1689, in imsave
    image.save(fname, **pil_kwargs)
  File "/home/ianni/.virtualenvs/PRISMER/lib/python3.8/site-packages/PIL/Image.py", line 2429, in save
    fp = builtins.open(filename, "w+b")
FileNotFoundError: [Errno 2] No such file or directory: 'helpers/labels/ocr_detection/helpers/images/COCO_test2015_000000000014_.png'
```
And this is what I get when I run accelerate launch experts/generate_ocrdet.py:

```
~/SW/prismer$ accelerate launch experts/generate_ocrdet.py
[19:23:05] WARNING  The following values were not passed to `accelerate launch` and had defaults used instead:                       launch.py:1088
                            `--num_processes` was set to a value of `1`                                                                            
                            `--num_machines` was set to a value of `1`                                                                             
                            `--mixed_precision` was set to a value of `'no'`                                                                       
                            `--dynamo_backend` was set to a value of `'no'`                                                                        
                    To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.                        
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:06<00:00,  6.05s/it]

```
That is, no error messages. Nothing.
I'm running python 3.8.10 and pytorch 2 with cuda 11.7:

```
python
Python 3.8.10 (default, Mar 13 2023, 10:26:41) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch;
>>> torch.__version__
'2.0.0+cu117'
>>> 
```
Is there any other info that you may need?

I've already checked the online demo, now I'm trying to connect your software with mine to produce some experimental results for my research.

**Comment by lorenmt** [comment]
This is strange. If there is no error message, then the model inference should be correct. So is there anything being saved in ocr_detection folder? The model however is asked to skip if it cannot detect any texts, but I remember there are two or three images in the example folders that have texts. Anyways, could you just quickly try 'python experts/generate_ocrdet.py' instead of 'accelerate launch?'

**Comment by ianni67** [comment]
I did it, got just the same output.
The folder ocr_detection is not there at all. Only the other experts created their folders.

```
~/SW/prismer/helpers/labels$ ls
depth  edge  normal  obj_detection  seg_coco
```

**Comment by lorenmt** [comment]
I think in that case, the only reason I can think of is the package version mismatch. 

I have now updated the requirements.txt to match the exact same packages used to build the huggingface demo. So could you install these packages and see whether it resolves the issue?

**Comment by ianni67** [comment]
I ran  
`pip install -r requirements.txt
`
and it downloaded a nymber of packages.
However, when I ran again the demo, nothing changed. I guess I'm doing something wrong, but don't know what... 
I do have some experience, but I'm missing the point with this issue.

I should add that running 
`python demo.py --exp_name prismer_base`

produces the expected labels, but without the help of the ocr.

**Comment by lorenmt** [comment]
Ok. Sorry I am not able to help more, as I am currently not able to reproduce your problems.

Maybe just restart with a new virtual environment or something. As I have received other people's feedback, saying they did not have any problems running the demo.

Well, since these expert labels are only considered as auxiliary knowledge, running Prismer with no OCR labels should still work. If you just need to work with some good vision-language models, PrismerZ without any experts is also pretty good.

**Comment by ianni67** [comment]
Thank you for your help.
I'll try with a new virtualenv.
Best regards
Giancarlo 

⁣Giancarlo Iannizzotto

Ottieni TypeApp per Android ​

Il giorno 29 apr 2023, 21:10, alle ore 21:10, Shikun Liu ***@***.***> ha scritto:

**Comment by ianni67** [comment]
Just as a follow-up, I'd add that I tried with an all-new virtualenv, but nothing changed. 
Moreover, I would add that the "normal" helper often (not for all images) requires more than 5GB of GPU memory, that made its use impractical for me (I have a 4GB nvidia card). As a consequence I disabled the "normal" helper. However the model works just well, indeed much better than other models (such as some transformers-based ones).
Thank you for sharing. If I will end up with publishing a scientific paper I will undoubtedly cite your work.
